### PR TITLE
LIME-315: Add Avatar in Header

### LINKED ISSUE #315: Add avatar component to header
- [ ] Use **avatar.tsx** instead of a hard-coded image in header 

Image

### PR DESCRIPTION
**Pull request by valery-chumak**

### CODE REVIEW

**Review by DavidCharif — DISMISSED** [review]

**Comment by valery-chumak** [comment]
image

File: frontend/src/bundles/common/components/header/header.tsx [hunk 1]
```diff
@@ -13,6 +14,10 @@ type HeaderProperties = {
 };
 
 const Header = ({ toggleSidebar }: HeaderProperties): JSX.Element => {
+    const { email, avatarUrl } = useAppSelector(({ auth }) => ({
+        email: auth.user?.email,
+        avatarUrl: auth.user?.avatarUrl,
+    }));
```
Comment (dimapopovych) [inline frontend/src/bundles/common/components/header/header.tsx]: just select whole user and cast it to UserDto with `as` so you don't need to write all of these optional chainings etc. You sure that user will be present here, TS doesn't

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/header/header.tsx [hunk 2]
```diff
@@ -14,6 +16,12 @@ type HeaderProperties = {
 
 const Header = ({ toggleSidebar }: HeaderProperties): JSX.Element => {
     const { theme } = useAppSelector((state) => state.theme);
```
Comment (stpalamar) [inline frontend/src/bundles/common/components/header/header.tsx]:
```suggestion
    const { theme } = useAppSelector(({ theme }) => theme);
```

File: frontend/src/bundles/common/components/header/header.tsx [hunk 3]
```diff
@@ -14,6 +16,12 @@ type HeaderProperties = {
...
+    const { user } = useAppSelector(({ auth }) => ({
+        user: auth.user as UserAuthResponseDto,
+    }));
```
Comment (stpalamar) [inline frontend/src/bundles/common/components/header/header.tsx]:
```suggestion
    const { user } = useAppSelector(({ auth }) => auth);
```

**Review by stpalamar — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/header/header.tsx (hunk 3 above)
Comment (stpalamar) [reply frontend/src/bundles/common/components/header/header.tsx]: Write useAppSelector like this to avoid warnings from redux

Comment (valery-chumak) [reply frontend/src/bundles/common/components/header/header.tsx]: Fixed. Thanks!

**Review by jhladkov — DISMISSED** [review]

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx [hunk 4]
```diff
@@ -25,10 +25,10 @@ const Navigation = ({ avatarUrl }: Properties): JSX.Element => {
...
                 </li>
                 <li>
-                    <img
-                        src={avatarUrl ?? defaultAvatar}
-                        alt="avatar"
-                        className={styles['avatar']}
+                    <Avatar
+                        size="sm"
+                        email={email}
+                        avatarUrl={avatarUrl || ''}
```
Comment (dimapopovych) [inline frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]:
```suggestion
  ...(avatarUrl && { avatarUrl })
```
if our linter will allows this)
but maybe will be better to just change typings to string | null

File: frontend/src/bundles/common/components/header/header.tsx [hunk 5]
```diff
@@ -13,7 +13,11 @@ type HeaderProperties = {
 };
 
 const Header = ({ toggleSidebar }: HeaderProperties): JSX.Element => {
-    const { theme } = useAppSelector((state) => state.theme);
+    const { theme } = useAppSelector(({ theme }) => theme);
+    const { user } = useAppSelector(({ auth }) => auth);
```
Comment (dimapopovych) [inline frontend/src/bundles/common/components/header/header.tsx]:
```suggestion
    const user = useAppSelector(({ auth }) => auth.user as UserAuthResponseDto);
```

File: frontend/src/bundles/common/components/header/header.tsx [hunk 6]
```diff
@@ -13,7 +13,11 @@ type HeaderProperties = {
...
+
+    const { email, avatarUrl } = user || {};
```
Comment (dimapopovych) [inline frontend/src/bundles/common/components/header/header.tsx]: and you can remove this part

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx (hunk 4 above)
Comment (valery-chumak) [reply frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]: It doesn't allow to do so( 
So, I changed the type of avatarUrl to string | null

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**